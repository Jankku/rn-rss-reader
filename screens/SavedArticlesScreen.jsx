import { useEffect, useState } from 'react';
import NewsList from '../components/Feed/NewsList';

function SavedArticlesScreen() {
  const [newsItems, setNewsItems] = useState([]);

  useEffect(() => {
    (async () => {
      const res = await FeedController.getFeedById(regionId);
      setNewsItems(res.rss.channel.item);
    })();
  }, [regionId]);

  return (
    <View style={{ flex: 1 }}>
      <NewsList data={newsItems} />
    </View>
  );
}

export default SavedArticlesScreen;
