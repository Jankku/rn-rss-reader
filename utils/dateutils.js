import { DateTime } from 'luxon';

/**
 * For articles released today return time and date otherwise.
 * @param {string} pubDate - RFC2822 date
 * @returns {string} Formatted date
 */
export function RFC2822ToTimeOrDate(pubDate) {
  const dateObject = DateTime.fromRFC2822(pubDate);
  const formattedDate =
    dateObject.toRelativeCalendar() === 'today'
      ? dateObject.toLocaleString(DateTime.TIME_24_SIMPLE)
      : dateObject.toLocaleString(DateTime.DATE_SHORT);
  return formattedDate;
}

/**
 * @param {string} pubDate - RFC2822 date
 * @returns {string} Formatted date in DateTime format
 */
export function RFC2822ToDateTime(pubDate) {
  const dateObject = DateTime.fromRFC2822(pubDate);
  const formattedDate = dateObject.toLocaleString(DateTime.DATETIME_SHORT);
  return formattedDate;
}
