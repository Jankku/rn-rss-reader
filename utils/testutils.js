import { render } from '@testing-library/react-native';
import { ThemeContextProvider } from '../context/ThemeContext';
import { LocationContextProvider } from '../context/LocationContext';
import { RegionContextProvider } from '../context/RegionContext';

const AllProviders = ({ children }) => {
  return (
    <LocationContextProvider>
      <RegionContextProvider>
        <ThemeContextProvider>{children}</ThemeContextProvider>
      </RegionContextProvider>
    </LocationContextProvider>
  );
};

const customRender = (ui, options) => render(ui, { wrapper: AllProviders, ...options });

// re-export everything
export * from '@testing-library/react-native';

// override render method
export { customRender as render };
