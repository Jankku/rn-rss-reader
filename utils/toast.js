import Toast from 'react-native-root-toast';

const showToast = (message) =>
  Toast.show(message, {
    position: 100,
    opacity: 0.9,
    shadow: false,
  });

export default showToast;
