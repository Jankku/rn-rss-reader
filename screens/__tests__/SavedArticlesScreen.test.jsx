import SavedArticlesScreen from '../SavedArticlesScreen';
import { render } from '../../utils/testutils';

describe('Test SavedArticlesScreen', () => {
  test('Should match snapshot', () => {
    const { toJSON } = render(
      <SavedArticlesScreen navigation={{ navigate: () => {}, setOptions: () => {} }} />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
