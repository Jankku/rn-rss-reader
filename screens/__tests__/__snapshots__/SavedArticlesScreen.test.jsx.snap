// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SavedArticlesScreen Should match snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    ItemSeparatorComponent={
      Object {
        "$$typeof": Symbol(react.memo),
        "compare": null,
        "type": [Function],
      }
    }
    ListEmptyComponent={
      <ListEmptyComponent
        text="Save some articles"
      />
    }
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "flexGrow": 1,
        "justifyContent": "center",
      }
    }
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <Text
        style={
          Object {
            "color": "#101010",
            "fontSize": 16,
            "fontWeight": "700",
            "textAlign": "center",
          }
        }
      >
        Save some articles
      </Text>
    </View>
  </RCTScrollView>
</View>
`;
